plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryResultPageSource.java

```diff
@@ -87,10 +86,10 @@ public BigQueryResultPageSource(
             BigQuerySplit split,
             List<BigQueryColumnHandle> columns)
     {
-        this.bigQueryStorageClient = requireNonNull(bigQueryStorageClientFactory, "bigQueryStorageClientFactory is null").createBigQueryStorageClient();
-        this.split = requireNonNull(split, "split is null");
+        this.bigQueryStorageClient = bigQueryStorageClientFactory.createBigQueryStorageClient();
```

Comment:
what harm did it make? Actually IMO it is better to do defensive checks during object construction unless not feasible due to perfomance reasons, even if current code structure suggest those are not necessary. Those can come up handy if we restructure the code later on, and allow to catch bugs early.